presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -866,6 +866,38 @@ public void testCreateUnpartitionedTableAsSelect()
         }
     }
 
+    @Test
+    public void testInsertIntoPartitionedTable()
+    {
+        // Generate temporary table name.
+        String tmpTableName = generateRandomTableName();
+        // Clean up if temporary table already exists.
+        dropTableIfExists(tmpTableName);
+
+        try {
+            getQueryRunner().execute(getSession(), String.format("CREATE TABLE %s (name VARCHAR, regionkey BIGINT, nationkey BIGINT) WITH (partitioned_by = ARRAY['regionkey','nationkey'])", tmpTableName));
+
+            // Test insert into an empty table.
+            Session writeSession = buildSessionForTableWrite();
+            getQueryRunner().execute(writeSession, String.format("INSERT INTO %s SELECT name, regionkey, nationkey FROM nation", tmpTableName));
+            assertQuery(String.format("SELECT * FROM %s", tmpTableName), "SELECT name, regionkey, nationkey FROM nation");
+
+            // Test failure on insert into existing partitions.
+            assertQueryFails(writeSession, String.format("INSERT INTO %s SELECT name, regionkey, nationkey FROM nation", tmpTableName),
+                    ".*Cannot insert into an existing partition of Hive table: regionkey=.*/nationkey=.*");
+
+            // Test insert into existing partitions if insert_existing_partitions_behavior is set to OVERWRITE.
+            Session overwriteSession = Session.builder(writeSession)
+                    .setCatalogSessionProperty("hive", "insert_existing_partitions_behavior", "OVERWRITE")
+                    .build();
+            getQueryRunner().execute(overwriteSession, String.format("INSERT INTO %s SELECT name, regionkey, nationkey FROM nation", tmpTableName));
```

Comment:
nit: perhaps, modify the query to insert different data, say name || '.test', to make sure the overwrite actually happens